core/task.go

```diff
@@ -157,13 +158,56 @@ type TaskErrors interface {
 
 type TaskCreationRequest struct {
 	Name        string            `json:"name"`
+	Version     int               `json:"version"`
 	Deadline    string            `json:"deadline"`
 	Workflow    *wmap.WorkflowMap `json:"workflow"`
 	Schedule    Schedule          `json:"schedule"`
 	Start       bool              `json:"start"`
 	MaxFailures int               `json:"max-failures"`
 }
 
+func (tr *TaskCreationRequest) UnmarshalJSON(data []byte) error {
+	t := make(map[string]json.RawMessage)
+	if err := json.Unmarshal(data, &t); err != nil {
+		return err
+	}
+	for k, v := range t {
+		switch k {
+		case "name":
+			if err := json.Unmarshal(v, &(tr.Name)); err != nil {
+				return fmt.Errorf("%v (while parsing 'name')", err)
+			}
+		case "deadline":
+			if err := json.Unmarshal(v, &(tr.Deadline)); err != nil {
+				return fmt.Errorf("%v (while parsing 'deadline')", err)
+			}
+		case "workflow":
+			if err := json.Unmarshal(v, &(tr.Workflow)); err != nil {
+				return err
+			}
+		case "schedule":
+			if err := json.Unmarshal(v, &(tr.Schedule)); err != nil {
+				return err
+			}
+		case "start":
+			if err := json.Unmarshal(v, &(tr.Start)); err != nil {
+				return fmt.Errorf("%v (while parsing 'start')", err)
+			}
+		case "max-failures":
+			if err := json.Unmarshal(v, &(tr.MaxFailures)); err != nil {
+				return fmt.Errorf("%v (while parsing 'max-failures')", err)
+			}
+		case "version":
+			if err := json.Unmarshal(v, &(tr.Version)); err != nil {
+				return fmt.Errorf("%v (while parsing 'version')", err)
+			}
+		default:
+			return fmt.Errorf("Unrecognized key '%v' in task creation request", k)
```

Comment:
Identifying the unrecognized key is great.
It's hard to know what are corrected keys.